lighthouse-core/audits/non-composited-animations.js

```diff
@@ -0,0 +1,135 @@
+  /** [ICU Syntax] Label identifying the number of animations that are not composited. */
+  displayValue: `{itemCount, plural,
+  =1 {# animated element found}
+  other {# animated elements found}
+  }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/**
+ * Each failure reason is represented by a bit flag. The bit shift operator '<<' is used to define which bit corresponds to each failure reason.
+ * https://source.chromium.org/search?q=f:compositor_animations.h%20%22enum%20FailureReason%22
+ * @type {{flag: number, text: string}[]}
+ */
+const ACTIONABLE_FAILURE_REASONS = [
+  {
+    flag: 1 << 13,
+    text: 'Unsupported CSS Property',
+  },
+];
+
+/**
+ * Return list of actionable failure reasons and a boolean if some reasons are not actionable.
+ * Each flag is a number with a single bit set to 1 in the position corresponding to a failure reason.
+ * We can check if a specific bit is true in the failure coding using bitwise and '&' with the flag.
+ * @param {number} failureCode
+ * @return {string[]}
+ */
+function getActionableFailureReasons(failureCode) {
+  return ACTIONABLE_FAILURE_REASONS
+    .filter(reason => failureCode & reason.flag)
+    .map(reason => reason.text);
+}
+
+class NonCompositedAnimations extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'non-composited-animations',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['TraceElements', 'HostUserAgent'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts) {
+    // COMPAT: This audit requires m86
+    const match = artifacts.HostUserAgent.match(/Chrome\/(\d+)/);
+    if (!match || Number(match[1]) < 86) {
+      return {
+        score: 1,
+        notApplicable: true,
+      };
+    }
+
+    /** @type LH.Audit.Details.TableItem[] */
+    const results = artifacts.TraceElements
+      .filter(element => element.traceEventType === 'animation')
+      .map(element => {
+        /** @type LH.Audit.Details.NodeValue */
+        const node = {
+          type: 'node',
+          path: element.devtoolsNodePath,
+          selector: element.selector,
+          nodeLabel: element.nodeLabel,
+          snippet: element.snippet,
+        };
+
+        return {
+          node,
+          failureReasons: '', // TODO: Use for node specific failure reasons (e.g. incompatible animations)
+          subItems: {
+            type: 'subitems',
+            items: (element.animations || [])
+              .filter(({failureReasonsMask}) => failureReasonsMask !== undefined)
+              .map(({name, failureReasonsMask}) => {
+                const failureStrings = getActionableFailureReasons(failureReasonsMask || 0);
```

Comment:
Vscode doesn't recognize when we filter out all animations with undefined `failureReasonsMask` using `filter()` so it throws a warning that `failureReasonsMask` may be undefined here.

I add this defensive logic to get rid of the warning.  Would it be better to just ignore the warnings in vscode?